src/frequenz/sdk/timeseries/_quantities.py

```diff
@@ -309,6 +310,55 @@ def __sub__(self, other: Self) -> Self:
         """
         return self.from_watts(self.as_watts() - other.as_watts())
 
+    def __mul__(self, duration: timedelta) -> "Energy":
```

Comment:
True, at least for some types,  I will get to it.

Reply:
I've added `from __future__ import annotations`.

Regarding the mul and truediv methods,  I'm not sure it would always be possible for all types,  no a general method is not a good idea.   

Also from a performance point of view,  this might be bad,  especially if we also want to support division by multiple types,  and we have to call `isinstance` 3 times to get to the right type.  And then `isinstance(v, float)` will reject ints, but the typing system will accept ints, so have to have an explicit runtime type check for ints also.

Actually I'm having second thoughts about this whole PR because of the performance costs.  Or at least about all arithmetic operations on these types.  Because we are encouraging code like 

```python
c = current1 + current2
```

but that line takes around 15 times as long as if it were just integers.

Reply:
> Regarding the mul and truediv methods, I'm not sure it would always be possible for all types, no a general method is not a good idea.

Why?

>  And then isinstance(v, float) will reject ints, but the typing system will accept ints, so have to have an explicit runtime type check for ints also.

Why wouldn't you allow `int`s?

> Also from a performance point of view

Yeah, this is a hard question besides this case in particular. Still, if we are already supporting several types for these operations, I guess one more or one less is not a huge deal. Just for the records, we discussed the performance and we concluded that for live data this should be fine, for historical data we might need to rethink the design in general.

How do you feel about supporting `float` or `int` ([`Real`](https://peps.python.org/pep-3141/) I would say) after that conversation?

Reply:
>> And then isinstance(v, float) will reject ints, but the typing system will accept ints, so have to have an explicit runtime type check for ints also.
>
> Why wouldn't you allow ints?

We would allow ints,  my comment above is about how we would have to have two `isinstance` checks, one for float and one for int.

So it would look like (for Power):

```python
    def __truediv__(self, other: Current | Voltage | float) -> Voltage | Current | Power:
        if isinstance(other, Current):
            return Voltage(self._base_value / other._base_value, exponent=0)
        if isinstance(other, Voltage):
            return Current(self._base_value / other._base_value, exponent=0)
        if isinstance(other, float) or isinstance(other, int):
            return type(self)(self._base_value / other, exponent=0)
        raise TypeError(
            f"unsupported operand type(s) for /: '{type(self)}' and '{type(other)}'"
        )

Reply:
> How do you feel about supporting float or int ([Real](https://peps.python.org/pep-3141/) I would say) after that conversation?

Is this about the solution you proposed during the call?  Can you explain it again with a code example?

Reply:
* https://github.com/frequenz-floss/frequenz-sdk-python/discussions/451